### ISSUE TITLE
False negative when using `typing.Self` in `@classmethod`

### ISSUE THREAD

**Issue by Lunarmagpie** [microsoft/pyright#4609]
**Describe the bug**
`@classmethod` returning `type[typing.Self]` is allowed to be typed as returning `typing.Self`. 

**To Reproduce**
```python
import typing

class Foo:
    @classmethod
    def bar(cls) -> typing.Self:
        return cls
```

**Expected behavior**
This should be a type error.

**VS Code extension or command-line**
Using command line.
**Additional context**
using pyright version 1.1.293

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.294, which I just published. It will also be included in a future release of pylance.
